extensions/integration-tests/src/schemaCompare.test.ts

```diff
@@ -31,146 +32,183 @@ if (context.RunTest) {
 				attempts--;
 				await utils.sleep(1000); // To ensure the providers are registered.
 			}
+			schemaCompareTester = new SchemaCompareTester();
 			console.log(`Start schema compare tests`);
 		});
 		test('Schema compare dacpac to dacpac comparison', async function () {
+			await schemaCompareTester.SchemaCompareDacpacToDacpac();
+		});
+		test('Schema compare database to database comparison and script generation', async function () {
+			await schemaCompareTester.SchemaCompareDatabaseToDatabase();
+		});
+		test('Schema compare dacpac to database comparison and script generation', async function () {
+			await schemaCompareTester.SchemaCompareDacpacToDatabase();
+		});
+	});
+}
+
+class SchemaCompareTester {
+	private static ParallelCount = 1;
+
+	@stressify({ dop: SchemaCompareTester.ParallelCount })
+	async SchemaCompareDacpacToDacpac(): Promise<void> {
+		assert(schemaCompareService, 'Schema Compare Service Provider is not available');
+
+		let source: azdata.SchemaCompareEndpointInfo = {
+			endpointType: azdata.SchemaCompareEndpointType.Dacpac,
+			packageFilePath: dacpac1,
+			serverName: '',
+			databaseName: '',
+			ownerUri: '',
+		};
+		let target: azdata.SchemaCompareEndpointInfo = {
+			endpointType: azdata.SchemaCompareEndpointType.Dacpac,
+			packageFilePath: dacpac2,
+			serverName: '',
+			databaseName: '',
+			ownerUri: '',
+		};
+
+		let schemaCompareResult = await schemaCompareService.schemaCompare('testOperationId', source, target, azdata.TaskExecutionMode.execute, null);
+		this.assertSchemaCompareResult(schemaCompareResult);
+	}
+
+	@stressify({ dop: SchemaCompareTester.ParallelCount })
+	async SchemaCompareDatabaseToDatabase(): Promise<void> {
+		let server = await getStandaloneServer();
+		await utils.connectToServer(server, SERVER_CONNECTION_TIMEOUT);
+
+		let nodes = <azdata.objectexplorer.ObjectExplorerNode[]>await azdata.objectexplorer.getActiveConnectionNodes();
+		assert(nodes.length > 0, `Expecting at least one active connection, actual: ${nodes.length}`);
+
+		let index = nodes.findIndex(node => node.nodePath.includes(server.serverName));
+		assert(index !== -1, `Failed to find server: "${server.serverName}" in OE tree`);
+
+		let ownerUri = await azdata.connection.getUriForConnection(nodes[index].connectionId);
+		let now = new Date();
+
+		let sourceDB: string = 'ads_schemaCompare_sourceDB_' + now.getTime().toString();
+		let targetDB: string = 'ads_schemaCompare_targetDB_' + now.getTime().toString();
+
+		try {
+			let dacfxService = await azdata.dataprotocol.getProvider<azdata.DacFxServicesProvider>('MSSQL', azdata.DataProviderType.DacFxServicesProvider);
+			assert(dacfxService, 'DacFx Service Provider is not available');
+			let result1 = await dacfxService.deployDacpac(dacpac1, sourceDB, true, ownerUri, azdata.TaskExecutionMode.execute);
```

Comment:
it would be good to make sure the databases get created using assertDatabaseCreationResult() (added in https://github.com/microsoft/azuredatastudio/pull/6049/files) since deployDacpac returns before the deploy is actually complete